core/src/main/java/com/topjohnwu/superuser/internal/SerialExecutorService.java

```diff
@@ -60,7 +45,7 @@ public synchronized void execute(Runnable r) {
         }
         mTasks.offer(r);
         if (scheduleTask == null) {
-            scheduleTask = new FutureTask<>(this);
+            scheduleTask = new FutureTask<>(this::executeTasks);
```

Comment:
I avoided this to prevent another object allocation, as on Android this will be desugared to a new class.
Exposing callable interface should be fine though, this is internal API and not designed for external usage.